### PR TITLE
21 prof reviews requester

### PR DESCRIPTION
**Pull request by susiirwin**
@JeanJoeris, @brendandillon, @rdavid1099 - i need your assistance please. It seems that the code is working as expected, but the feature test for this is not working. Once this is working, i believe we can refactor this in such a way that both Professionals and Requesters can use this same review form.

### CODE REVIEW

File: app/views/requesters/reviews/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+<%= form_for @review, url: requesters_reviews_path do |f| %>
+
+  <%= f.hidden_field :requester_id, value: @review.requester_id %>
+  <%= f.hidden_field :professional_id, value: @review.professional_id %>
+  <%= f.hidden_field :reviewee_role, value: @review.reviewee_role %>
```
Comment (JeanJoeris) [inline app/views/requesters/reviews/new.html.erb:6]: I changed these to f.hidden_field to have them nested inside review in params

File: spec/features/professional_creates_review_for_requester_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+      expect(current_path).to eq(new_requesters_review_path)
+
+      fill_in 'review_review', with: 'This is my review of the requester'
+      choose('review_rating_four_stars')
+
+      click_on "Create Review"
+
+      review = Review.last
+      expect(current_path).to eq (requesters_review_path(review))
```
Comment (JeanJoeris) [inline spec/features/professional_creates_review_for_requester_spec.rb:25]: I changed `@review` to `review`

File: app/controllers/requesters/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  def show
+    @review = Review.find(params[:id])
+  end
+
+  private
+
+    def review_params
+      params.require(:review).permit(:review, :rating, :professional_id, :requester_id, :reviewee_role)
```
Comment (JeanJoeris) [inline app/controllers/requesters/reviews_controller.rb:25]: `:reviewee_role` wasn't in the strong params

**Review by JeanJoeris — COMMENTED** [review]
This looks great and only needed a few tiny tweaks to be working

**Comment by susiirwin** [comment]
Do we want to look at using this for Requesters writing reviews before we merge?

File: app/controllers/requesters/reviews_controller.rb (hunk 3 above)
Comment (susiirwin) [reply app/controllers/requesters/reviews_controller.rb:25]: thank you!!!

File: spec/features/professional_creates_review_for_requester_spec.rb (hunk 2 above)
Comment (susiirwin) [reply spec/features/professional_creates_review_for_requester_spec.rb:25]: yup! makes total sense NOW. sigh.

File: app/views/requesters/reviews/new.html.erb (hunk 1 above)
Comment (susiirwin) [reply app/views/requesters/reviews/new.html.erb:6]: interesting.... i am sort of surprised this worked, but I will believe it.

File: app/models/user.rb [hunk 4]
```diff
@@ -37,6 +37,17 @@ def closed_jobs
...
 
+  def inverse_role
+    inverse_roles[role]
+  end
+
+  def inverse_roles
+    {
+      "professional" => "requester",
+      "requester" => "professional"
+    }
```
Comment (JeanJoeris) [inline app/models/user.rb:61]: Very readable. I know exactly what it does and how it works at a glance

**Comment by susiirwin** [comment]
I have squashed the merge conflicts. This is good to go now :)

**Pull request merged**